applications/trilinos_application/custom_utilities/trilinos_dof_updater.h

```diff
@@ -0,0 +1,265 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Jordi Cotela
+//
+
+#if !defined(KRATOS_TRILINOS_DOF_UPDATER_H_INCLUDED )
+#define  KRATOS_TRILINOS_DOF_UPDATER_H_INCLUDED
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "utilities/dof_updater.h"
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Utility class to update the values of degree of freedom (Dof) variables after solving the system.
+/** This class encapsulates the operation of updating nodal degrees of freedom after a system solution.
+ *  In pseudo-code, the operation to be performed is
+ *  for each dof: dof.variable += dx[dof.equation_id]
+ *  This operation is a simple loop in shared memory, but requires additional infrastructure in MPI,
+ *  to obtain out-of-process update data. TrilinosDofUpdater manages both the update operation and the
+ *  auxiliary infrastructure.
+ */
+template< class TSparseSpace >
+class TrilinosDofUpdater : public DofUpdater<TSparseSpace>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of TrilinosDofUpdater
+    KRATOS_CLASS_POINTER_DEFINITION(TrilinosDofUpdater);
+
+    using BaseType = DofUpdater<TSparseSpace>;
+    using DofsArrayType = typename BaseType::DofsArrayType;
+    using SystemVectorType = typename BaseType::SystemVectorType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    TrilinosDofUpdater():
+        DofUpdater<TSparseSpace>()
+    {}
+
+    /// Deleted copy constructor
+    TrilinosDofUpdater(TrilinosDofUpdater const& rOther) = delete;
+
+    /// Destructor.
+    ~TrilinosDofUpdater() override {}
```

Comment:
General question: why in kratos do we define a do-nothing destructor when it is not a virtual base destructor? Is it not the same as the default?

Reply:
It is, unless you eventually derive some class from it and forgot to add it. Also having it is not harmless and avoid such bugs.